chainerx_cc/chainerx/cuda/cuda_device/linalg.cu

```diff
@@ -204,5 +284,142 @@ public:
 
 CHAINERX_CUDA_REGISTER_KERNEL(InverseKernel, CudaInverseKernel);
 
+class CudaSVDKernel : public SVDKernel {
+public:
+    std::tuple<Array, Array, Array> Call(const Array& a, bool full_matrices, bool compute_uv) override {
+        Device& device = a.device();
+        Dtype dtype = a.dtype();
+        CudaSetDeviceScope scope{device.index()};
+
+        CHAINERX_ASSERT(a.ndim() == 2);
```

Comment:
Could you support batched SVD?
https://docs.nvidia.com/cuda/cusolver/index.html#cuds-lt-t-gt-gesvdjbatch

Reply:
The plan is to implement things only for matrices first. Batched version of SVD won't be included in this PR.